lib/collection/request.js

```diff
@@ -107,6 +107,38 @@ _.assign(Request.prototype, /** @lends Request.prototype */ {
     },
 
     /**
+     * Sets authentication method for the request
+     *
+     * @param {?String|RequestAuth~definition} type
+     * @param {VariableList~definition=} [options]
+     *
+     * @note This function was previously (in v2 of SDK) used to clone request and populate headers. Now it is used to
+     * only set auth information to request
+     *
+     * @note that ItemGroup#authorizeUsing depends on this function
+     */
+    authorizeUsing: function (type, options) {
```

Comment:
@shamasis Two points,

1. This API looks weird, we would have to do 
```
item.request.authorizeUsing(item.getAuth());
auth = item.getAuth();
```
`auth` is set on a request, but get from an item. `authorizeUsing` does not return the `auth`. So having to do `context.item.getAuth()` again.

2. We set the parent of the auth to the request, but the parent of request is not set to the item 😕

Can we do `auth.setParent(this)` in `item.getAuth` ?

Reply:
1. This function is meant to set auth in a group or item. It is not for getting auth from somewhere to set here. It is more like functions such as .setHeaders, etc.

2. Then we should bypass request and set parent to item. That would be weird though.

Reply:
Can we set the parent of the request to the item?

Reply:
Added reference resolution to `.toObjectResolved` instead.